### ISSUE TITLE
In some specific cases 412/Precondition Failed responses are generated for valid ETags

### ISSUE THREAD

**Issue by gcotelli** [ba-st/Stargate#164]
When using `createEntityTagHashingEncodedResource` as the ETag calculation strategy and `from:within:get:thenUpdateWith:` to implement a route and the controller support several media types; if the client requesting the update doesn't send an `Accept` header with the same media type used as `Content-Type` the response will be `412/Precondition Failed` even if the provided ETag in the `If-Match` header is correct.

This happens because the method checking the pre-condition is encoding the resource using the target media type (the one negotiated by the `Accept` header) instead of using the one in `Content-Type`.

When fixing this problem we need to take care to not affect the other uses of the ETag calculation (it's used during response generation). The only sender using it not for generating a response is this update method.

In summary, to reproduce the problem all of this needs to happen at the same time:
- `createEntityTagHashingEncodedResource` is used as ETag calculation strategy
- the controller supports several media types
- the controller implements an update endpoint using `from:within:get:thenUpdateWith:`
- the client doesn't send an `Accept` header matching the `Content-Type` one.
